### PR TITLE
Address surfaceProperty:ExteriorNaturalVentedCavity results in convection model errors issue

### LINKED ISSUE #9901: SurfaceProperty:ExteriorNaturalVentedCavity results in convection model errors
Issue overview
--------------

When applying the **SurfaceProperty:ExteriorNaturalVentedCavity** object to roof surfaces to simulate the ventilated air cavity between the roof tiles and the underlying surfaces, it results in convection model errors. This happens when the AdaptiveConvectionAlgorithm is selected in the **SurfaceConvectionAlgorithm:Outside** object, and if the following roof convection models are chosen in the **SurfaceConvectionAlgorithm:Outside:AdaptiveModelSelections** object (field Wind Convection Horizontal Roof Equation Source):

Model -> Error
ClearRoof (default) -> CalcClearRoof: Convection model not evaluated (bad value for distance to roof edge)
TARPWindward -> CalcSparrowWindward: Convection model not evaluated (bad face area)
BlockenWindward -> CalcBlockenWindward: Convection model wind angle calculation suspect (developer issue)

Other models and algorithms work fine.

An IDF file with the error example is attached along with the specific weather file. However, the error is not dependent on the weather conditions/file.

### Details
Some additional details for this issue (if relevant):
 - Windows 11 64-bit (10.0.22621) & Mac OS 13.1 (22C65)
 - EnergyPlus v22.2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

[Files.zip](https://github.com/NREL/EnergyPlus/files/11017592/Files.zip)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #9901
 - This fix removes false severe error message when the exterior boundary condition of a roof surface is `OtherSideCondModeled` and the exterior convection algorithm (SurfaceConvectionAlgorithm:Outside) specified is `AdaptiveConvectionAlgorithm`. 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 1]
```diff
@@ -8812,16 +8812,18 @@ Real64 CalcClearRoof(EnergyPlusData &state,
...
+            if (state.dataConvectionCoefficient->CalcClearRoofErrorIDX == 0) {
+                ShowSevereMessage(state, "CalcClearRoof: Convection model not evaluated (bad value for distance to roof edge)");
+                ShowContinueError(state, format("Value for distance to roof edge ={:.3R}", x));
+                ShowContinueError(state, format("Occurs for surface named = {}", state.dataSurface->Surface(SurfNum).Name));
+                ShowContinueError(state, "Convection surface heat transfer coefficient set to 9.999 [W/m2-K] and the simulation continues");
+            }
+            ShowRecurringSevereErrorAtEnd(
+                state,
+                "CalcClearRoof: Convection model not evaluated because bad value for distance to roof edge and set to 9.999 [W/m2-k]",
+                state.dataConvectionCoefficient->CalcClearRoofErrorIDX);
```
Comment (Nigusse) [inline src/EnergyPlus/ConvectionCoefficients.cc:8825]: Now the error message is reported only when the exterior boundary condition of a surface is not `OtherSideCondModeledExt` because this boundary condition does not use the convection coefficient calculated here.

File: tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc [hunk 2]
```diff
@@ -3689,3 +3689,96 @@ TEST_F(ConvectionCoefficientsFixture, testTARPNaturalConvectionAlgorithm)
...
+        // Ext boundary conditions set to OtherSideCondModeledExt and expect no error
+        state->dataSurface->Surface(1).ExtBoundCond = EnergyPlus::DataSurfaces::OtherSideCondModeledExt;
+        RoofArea = 0.0;
+        RoofPerimeter = 0.0;
+        Real64 Hf_result3 = CalcClearRoof(*state, SurfNum, SurfTemp, AirTemp, WindAtZ, WindDirect, RoofArea, RoofPerimeter);
+        EXPECT_DOUBLE_EQ(9.9999, Hf_result3);
+        EXPECT_EQ(state->dataSurface->Surface(1).ExtBoundCond, EnergyPlus::DataSurfaces::OtherSideCondModeledExt);
+        EXPECT_TRUE(compare_err_stream(""));
+    }
+}
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc:3782]: Added unit test that demonstrates the error messages expected depending on the boundary condition of an exterior surface.

**Review by Nigusse — COMMENTED** [review]
Code change walk-though

**Comment by Nigusse** [comment]
A modified defect file is uploaded on defect files repository on GitHub, `\EnergyPlusDevSupport\DefectFiles\9000s\9901`.

**Comment by Nigusse** [comment]
Develop throws the following severe warning message when the exterior boundary condition of a roof surface is `OtherSideCondModeledExt` or `ExternalEnvironment` and the exterior convection algorithm (SurfaceConvectionAlgorithm:Outside) is AdaptiveConvectionAlgorithm.

```
   ** Severe  ** CalcClearRoof: Convection model not evaluated (bad value for distance to roof edge)
   **   ~~~   ** Value for distance to roof edge =0.000
   **   ~~~   ** Occurs for surface named = ZN_11_ATTIC_ROOF_1
   **   ~~~   ** Convection surface heat transfer coefficient set to 9.999 [W/m2-K] and the simulation continues

   *************  ** Severe  ** CalcClearRoof: Convection model not evaluated because bad value for distance to roof edge and set to 9.999 [W/m2-k]
   *************  **   ~~~   **   This error occurred 437814 total times;
   *************  **   ~~~   **   during Warmup 17334 times;
   *************  **   ~~~   **   during Sizing 219474 times.
   *************
```
This error message should not have been generated when the exterior boundary condition of a roof surface is `OtherSideCondModeledExt`. The fix eliminates this severe error message because it is not applicable.

File: tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc (hunk 2 above)
Comment (Myoldmopar) [reply tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc:3782]: This is a **fantastic** unit test @Nigusse, great work!  Clear and obvious demonstration of the problem with multiple scenarios, not a bunch of IDF getting in the way, good stuff!

File: src/EnergyPlus/ConvectionCoefficients.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/ConvectionCoefficients.cc:8825]: This fix makes perfect sense, you wouldn't want to emit an error for that case.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
CI is happy, and everything looks good here, merging.
